js/main.js

```diff
@@ -1 +1,59 @@
+(function () {
+  function debounce(func, wait, immediate) {
+  	var timeout;
+  	return function() {
+  		var context = this, args = arguments;
+  		var later = function() {
+  			timeout = null;
+  			if (!immediate) func.apply(context, args);
+  		};
+  		var callNow = immediate && !timeout;
+  		clearTimeout(timeout);
+  		timeout = setTimeout(later, wait);
+  		if (callNow) func.apply(context, args);
+  	};
+  };
 
+  function getElementByClassName(className) {
+    if (document.getElementsByClassName) {
+      return document.getElementsByClassName(className)[0];
+    }
+
+    if (document.querySelectorAll) {
+      return document.querySelectorAll((' ' + className).replace(/ +/g, '.'))[0];
+    }
+  };
+
+  function addClass(elem, className) {
+    var re = new RegExp(className, 'g');
+    var elemClassName = elem.className;
+
+    if (re.test(elemClassName) === false) {
+      elem.className = elemClassName + ' ' + className;
+    }
+  }
+
+  function removeClass(elem, className) {
+    var re = new RegExp(className, 'g');
+    var elemClassName = elem.className;
+
+    if (re.test(elemClassName) === true) {
+      elemClassName = elemClassName.replace(className, '');
+      // remove not need one space left after replacing `className`
+      elem.className = elemClassName.slice(0, elemClassName.length - 1);
+    }
```

Comment:
I'd find node.classList polyfill than writing custom function for add/remove class